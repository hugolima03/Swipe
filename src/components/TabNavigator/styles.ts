import styled, { css } from 'styled-components'

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;

    height: 6rem;

    gap: 7rem;

    background-color: ${theme.colors.white};
    box-shadow: ${theme.shadows.tabNavigator};

    border-radius: 1.6rem 1.6rem 0 0;

    position: relative;
  `}
`

export const TabLink = styled.a`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    color: ${theme.colors.black};
    font-weight: ${theme.font.bold};
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.xsmall};

    text-decoration: none;

    svg {
      height: 2.4rem;
      fill: ${theme.colors.black};
    }
  `}
`

export const HomeLink = styled.a`
  ${({ theme }) => css`
    display: flex;

    flex-direction: column;
    align-items: center;
    justify-content: center;

    width: 6.6rem;
    height: 6.6rem;
    border-radius: 50%;

    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -60%);

    background: ${theme.gradients.blue};

    text-decoration: none;

    color: ${theme.colors.black};
    font-family: ${theme.font.family};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.xsmall};
    color: white;

    svg {
      height: 2.4rem;
      fill: white;
    }
  `}
`
