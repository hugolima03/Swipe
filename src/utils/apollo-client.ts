import { ApolloClient, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: `${process.env.NEXT_PUBLIC_API_URL}`,
  cache: new InMemoryCache(),
  headers: { Authorization: `Bearer ${process.env.NEXT_PUBLIC_GRAPHCMS_TOKEN}` }
})

export default client
