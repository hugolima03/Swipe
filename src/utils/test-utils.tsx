import { MockedProvider, MockedResponse } from '@apollo/client/testing'

import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const mocks: MockedResponse<Record<string, unknown>>[] = []

const customRender = (
  ui: React.ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </MockedProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
