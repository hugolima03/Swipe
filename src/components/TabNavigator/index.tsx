import Link from 'next/link'
import { Home, School, Search } from 'styled-icons/material-rounded'
import * as S from './styles'

const TabNavigator = () => (
  <S.Wrapper>
    <Link href="/explore" passHref>
      <S.TabLink>
        <Search />
        Explore
      </S.TabLink>
    </Link>

    <Link href="/" passHref>
      <S.HomeLink>
        <Home />
        Home
      </S.HomeLink>
    </Link>

    <Link href="/profile" passHref>
      <S.TabLink>
        <School />
        Profile
      </S.TabLink>
    </Link>
  </S.Wrapper>
)

export default TabNavigator
