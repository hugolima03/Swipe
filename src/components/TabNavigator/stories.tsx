import { Story, Meta } from '@storybook/react'
import TabNavigator from '.'

export default {
  title: 'TabNavigator',
  component: TabNavigator,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: Story = () => (
  <div
    style={{
      height: '100vh',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'end'
    }}
  >
    <TabNavigator />
  </div>
)
