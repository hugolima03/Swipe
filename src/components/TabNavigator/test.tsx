import { render, screen } from 'utils/test-utils'
import TabNavigator from '.'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('<TabNavigator />', () => {
  it('should render the links corretcly', () => {
    useRouter.mockImplementation(() => ({
      pathname: '/'
    }))

    render(<TabNavigator />)

    expect(screen.getByRole('link', { name: /Explore/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Home/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Profile/i })).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /Home/i }).parentElement
    ).toMatchSnapshot()
  })
})
