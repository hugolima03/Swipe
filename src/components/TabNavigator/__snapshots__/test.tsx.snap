// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabNavigator /> should render the links corretcly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  height: 6rem;
  gap: 7rem;
  background-color: #FFF;
  box-shadow: rgba(25,1,52,0.16) 0px 4px 10px;
  border-radius: 1.6rem 1.6rem 0 0;
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #1E252B;
  font-weight: 700;
  font-family: Nunito Sans,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 svg {
  height: 2.4rem;
  fill: #1E252B;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 6.6rem;
  height: 6.6rem;
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-60%);
  -ms-transform: translate(-50%,-60%);
  transform: translate(-50%,-60%);
  background: linear-gradient(90.38deg,#3F37C9 -11.12%,#1BEAFF 110.38%);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #1E252B;
  font-family: Nunito Sans,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 700;
  font-size: 1.2rem;
  color: white;
}

.c3 svg {
  height: 2.4rem;
  fill: white;
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="/explore"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="24"
        width="24"
      />
      <path
        d="M20.29 18.88l-5.56-5.56c1.13-1.55 1.63-3.58.98-5.74-.68-2.23-2.57-3.98-4.85-4.44a6.511 6.511 0 00-7.72 7.72c.46 2.29 2.21 4.18 4.44 4.85 2.16.65 4.19.15 5.74-.98l5.56 5.56a.996.996 0 101.41-1.41zM5 9.5C5 7.01 7.01 5 9.5 5S14 7.01 14 9.5 11.99 14 9.5 14 5 11.99 5 9.5z"
      />
    </svg>
    Explore
  </a>
  <a
    class="c3"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M10.8 3.9l-6 4.5c-.5.38-.8.97-.8 1.6v9c0 1.1.9 2 2 2h4v-7h4v7h4c1.1 0 2-.9 2-2v-9c0-.63-.3-1.22-.8-1.6l-6-4.5a2.01 2.01 0 00-2.4 0z"
      />
    </svg>
    Home
  </a>
  <a
    class="c1"
    href="/profile"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
      >
        <rect
          height="24"
          width="24"
        />
        <rect
          height="24"
          width="24"
        />
      </g>
      <path
        d="M5 13.18v2.81c0 .73.4 1.41 1.04 1.76l5 2.73c.6.33 1.32.33 1.92 0l5-2.73c.64-.35 1.04-1.03 1.04-1.76v-2.81l-6.04 3.3c-.6.33-1.32.33-1.92 0L5 13.18zm6.04-9.66l-8.43 4.6c-.69.38-.69 1.38 0 1.76l8.43 4.6c.6.33 1.32.33 1.92 0L21 10.09V16c0 .55.45 1 1 1s1-.45 1-1V9.59c0-.37-.2-.7-.52-.88l-9.52-5.19a2.04 2.04 0 00-1.92 0z"
      />
    </svg>
    Profile
  </a>
</nav>
`;
